### PR TITLE
tests: filter spread logs to be sent to grafana

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Use log-filter tool to generate a .filtered.log file which is sent to grafana.

Those filtered logs don't include the error and debug output, just the executed tests and the results.

### CODE REVIEW

**Review by olivercalder — APPROVED** [review]
Looks good to me, thanks!

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -793,23 +821,12 @@ jobs:
...
-                  done
-              done
-          fi
           # Run spread tests
           # "pipefail" ensures that a non-zero status from the spread is
           # propagated; and we use a subshell as this option could trigger
           # undesired changes elsewhere
           echo "Running command: $SPREAD $RUN_TESTS"
-          (set -o pipefail; $SPREAD -no-debug-output -logs .logs $RUN_TESTS | tee spread.log)
+          (set -o pipefail; $SPREAD -no-debug-output -logs .logs $RUN_TESTS | ./utils/log-filter $FILTER_PARAMS | tee spread.log)
```
Comment (olivercalder) [inline .github/workflows/test.yaml:829]: Spread tests are failing, seems like maybe this should be:

```suggestion
          (set -o pipefail; $SPREAD -no-debug-output -logs .logs $RUN_TESTS | ./tests/lib/external/snapd-testing-tools/utils/log-filter $FILTER_PARAMS | tee spread.log)
```

**Review by olivercalder — CHANGES_REQUESTED** [review]
Problem with seeing the `log-filter` script in spread tests.

**Review by olivercalder — APPROVED** [review]
Thanks!

**Review by andrewphelpsj — APPROVED** [review]
Thanks!

**Pull request merged**